applicationinsights-react-js/src/ReactPlugin.ts

```diff
@@ -25,30 +32,33 @@ export default class ReactPlugin extends BaseTelemetryPlugin {
         let _extensionConfig: IReactExtensionConfig;
         let _unlisten: any;
         let _pageViewTimer: any;
+        let _pageViewTracked:boolean;
 
         dynamicProto(ReactPlugin, this, (_self, _base) => {
             _initDefaults();
 
             _self.initialize = (config: IConfiguration & IConfig, core: IAppInsightsCore, extensions: IPlugin[], pluginChain?:ITelemetryPluginChain) => {
                 super.initialize(config, core, extensions, pluginChain);
-                _extensionConfig =
-                    config.extensionConfig && config.extensionConfig[_self.identifier]
-                        ? (config.extensionConfig[_self.identifier] as IReactExtensionConfig)
-                        : { history: null };
-        
-                arrForEach(extensions, ext => {
-                    const identifier = (ext as ITelemetryPlugin).identifier;
-                    if (identifier === 'ApplicationInsightsAnalytics') {
-                        _analyticsPlugin = (ext as any) as IAppInsights;
+
+                _self._addHook(onConfigChange(config, (details) => {
+                    let ctx = _self._getTelCtx();
+                    _extensionConfig = ctx.getExtCfg<IReactExtensionConfig>(this.identifier, defaultReactExtensionConfig);
+                    _analyticsPlugin = core.getPlugin<any>(AnalyticsPluginIdentifier)?.plugin as IAppInsights;
+                    
+                    if (isFunction(_unlisten)) {
+                        _unlisten();
```

Comment:
Oh, interesting... I was not expecting this.

But this has identified a couple of potential issues that we should handle
- Once we call `_unlisten()` we should reset _unlisten to null
- Where we are "assigning" `_unlisten = null` we should check and call the function (so we don't leave any listeners behind).

Reply:
Yes, I forget to add set back to null here. The only other place we set unlisten to null is in initDefault, at where the listenr should not be existed so I think it should be good to go there